atest/robot/test_libraries/as_listener_global_vars.robot

```diff
@@ -0,0 +1,22 @@
+*** Settings ***
+Test Setup       Run Tests    ${EMPTY}    test_libraries/as_listener/global_vars_listener.robot
+Resource         atest_resource.robot
+
+*** Test Cases ***
+Check global variables in 'close' listener method
+    Stderr Should Be Equal To    SEPARATOR=\n
+        ...  Suite name:Global Vars Listener
+        ...  Suite documentation:Test global variables in 'close' listener method
+        ...  Previous test name:
+        ...  Previous test status:
+        ...  Log level:INFO
+        ...  Suite name:Global Vars Listener
+        ...  Suite documentation:Test global variables in 'close' listener method
+        ...  Previous test name:Global variables test
+        ...  Previous test status:PASS
+        ...  Log level:INFO
+        ...  Suite name:Global Vars Listener
+        ...  Suite documentation:Test global variables in 'close' listener method
+        ...  Previous test name:Global variables final test
+        ...  Previous test status:PASS
+        ...  Log level:INFO\n
```

Comment:
This test verifies that global variables are logged three times. I assume that's correct because the executed suite has two tests, and the library that has test case scope goes out of the scope (i.e. is closed) once per test and once also at the end of the test suite. This could still be made more explicit, and test coverage enhanced in general, like this:

1. Rename this test to something like `Global variables in listener 'close' method when library has test case scope`.

2. Add two more tests for libraries using the test suite and global scope. This requires also new libraries, but it's easy to create them so that they just extend the existing and set `ROBOT_LIBRARY_SCOPE` accordingly. It's possible to have all libraries in a single module, but that requires imports like `Library    module.Library` and that requires putting the directory containing them into `PYTHONPATH`. Might be easiest just to have separate files for each library.

3. Specify a set of variables that are logged as a list variable and then use it in each test as many time as needed.

Reply:
Created two more tests for the other scopes, however I'm not very happy with the naming of the new files/tests. If you @pekkaklarck find better suggestions I will update the PR.